src/python/pants/backend/python/goals/setup_py.py

```diff
@@ -862,6 +862,15 @@ async def get_requirements(
     direct_deps_tgts = await MultiGet(
         Get(Targets, DependenciesRequest(tgt.get(Dependencies))) for tgt in owned_by_us
     )
+    direct_deps_chained = OrderedSet(itertools.chain.from_iterable(direct_deps_tgts))
+    # Issue #17593: If ET has an undeclared requirement on R which has been resolved with an
```

Comment:
I didn't quite grok the comment at first (partly because of the naming), would you be open to this rewording?

"
If an python_requirement T has an undeclared requirement R, we recommend fixing that by adding an explicit dependency from T to a python_requirement target for R. In that case we want to represent these explicit deps in T's distribution metadata. See issue #17593.
"